### ISSUE TITLE
Can't view a file directly in the browser, it is always downloaded

### ISSUE THREAD

**Issue by Maofree** [PrestaShop/PrestaShop#31602]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hi
today a customer warned me on the fact that the pdf files that he had placed in a server folder, in the webroot of the site, he could not view them from the browser but made them download. Instead in the other site made with wordpress it was possible.
I checked the chrome settings and they are set to view pdf files in the browser.
I also put the pdf file on other sites with other wordpress and grav cms and joomla and from all I can view the pdfs in the browser.
then I put the pdf file on other sites made with prestashop and they all make me download the pdf when I go to type the url to the file in the browser

could be some information in the htaccess files used by prestashop

wordpress
https://www.progetto-group.it/qr/smaltimento/EAD_lampade.pdf

prestashop
https://www.bathman.it/QRCODE/EAD_lampade.pdf

with firefox I can see it on the browser, before it download the file then shows the file to the browser
with chrome it download the file, but just when is used prestashop

bye

### Steps to reproduce

1. upload a pdf file in the webroot of a site with prestashop
2. set the file's url to chrome's browser

### PrestaShop version(s) where the bug happened

1.7.8.8, 8.01

### PHP version(s) where the bug happened

7.4

**Comment by Hlavtox** [comment]
@maofree Bad server settings. Works on all my prestashop stores, example - https://www.trendo.cz/dokumenty/Schock_Akcni_sety_2022.pdf

**Comment by Maofree** [comment]
I don't think it's a server problem, because I've tried it on 4-5 different hosting servers and it always lets me download it. while on all the other cms that have similar servers it lets me view them. But it is true from your visualization

https://www.aloiserw.com/EAD_lampade.pdf

**Comment by Hlavtox** [comment]
Two different hostings:
- active24.cz/websupport.sk - https://deskyhevea.cz/dokumenty/hevea_2022_cz.pdf
- czechia.com - https://www.veran.cz/eshop/dokumenty/odstoupeni_od_kupni_smlouvy_cz.pdf

**Comment by Maofree** [comment]
veran is not with prestashop
with other cms there is no problem

**Comment by Hlavtox** [comment]
It is, in `/eshop` folder. :-) All my clients are on 1.7.6.9 if it helps. I tried to compare htaccess with 8.0.1 and there are some differences:

![Snímek obrazovky 2023-03-02 132635](https://user-images.githubusercontent.com/6097524/222428706-63560fa5-1711-448a-8e51-4c9f939451a5.jpg)

**Comment by Hlavtox** [comment]
The problem is in the htacccess, on 8.0.1 it also downloads the PDF instead of displaying it.

**Comment by Maofree** [comment]
yes it was the first thing I thought of, the htaccess file, because the other cms have it very basic
I'm testing with 1.7.8.8

**Comment by Maofree** [comment]
can it be due from this code?

    <FilesMatch "\.pdf$">
      Header set Content-Disposition "Attachment"
      Header set X-Content-Type-Options "nosniff"
    </FilesMatch>

**Comment by Maofree** [comment]
I've fixed this problem, set inside a folder a new htaccess with this code

`<FilesMatch "\.pdf$">
  Header set Content-Disposition "inline"
</FilesMatch>`


https://www.bathman.it/QRCODE/EAD_lampade.pdf

**Comment by Hlavtox** [comment]
@maofree It should respect the `download` attribute like this:

```
<a href="/images/myw3schoolsimage.jpg" download>
  <img src="/images/myw3schoolsimage.jpg" alt="W3Schools">
</a>
```

**Comment by Maofree** [comment]
but in this case the customer need to view in the browser the pdf
however in prestashop with Header set Content-Disposition "Attachment" will download always the pdf

perhaps being an ecommerce it needs a different behavior

**Comment by HanaRebaiQA** [comment]
Hello @maofree 

Thanks for your report and @Hlavtox for your feedback!

I was unable to reproduce your issue using the same steps to reproduce on PS8.0.1. 

https://user-images.githubusercontent.com/92912932/222474406-36a33886-06fc-4879-a3f0-a779daf3be80.mp4

Did I miss something?

Thank You!

**Comment by Hlavtox** [comment]
@HanaRebaiQA You are opening a local PDF file :D

**Comment by florine2623** [comment]
I think this should be considered as an improvement ^^

ping @PrestaShop/product-team wdyt

**Comment by Hlavtox** [comment]
@florine2623 It's a regression. Prestashop should not interfere with the ability to view or download files.

**Comment by Hlavtox** [comment]
@maofree Broken in this commit - https://github.com/PrestaShop/PrestaShop/pull/21143/commits/224d16f315f264f2b1c952ce59705c97fd6882f0

**Comment by Hlavtox** [comment]
@maofree Will be fixed by https://github.com/PrestaShop/PrestaShop/pull/31637

**Comment by Maofree** [comment]
thanks

**Comment by ShaiMagal** [comment]
@hibatallahAouadni i think, this can be closed.
Already fixed by #31637 (merged)
